File: redsheet-portal/src/app/database-models/project-background.ts
import { required, maxLength, range, nested } from '@rx/annotations';
import { NegotiationType, ProjectModule, RelationshipRequire, ValueObjective, LongTermObjective, BackgroundEvent, } from './'
export class ProjectBackground {
    constructor(projectBackground?: ProjectBackground) {
        let properties = ["focus", "opponentName", "projectBackgroundId", "reason", "negotiationTypeId", "projectModuleId", "relationshipRequireId", "valueObjectiveId", "longTermObjectives", "backgroundEvents",];
        for (let property of properties)
            if (projectBackground && projectBackground[property])
                this[property] = projectBackground[property];
    }

    @required()
    @maxLength(200)
    focus: string = undefined;

    @required()
    @maxLength(100)
    opponentName: string = undefined;

    projectBackgroundId: number = 0;

    @required()
    @maxLength(200)
    reason: string = undefined;

    @range(0, 2147483647)
    negotiationTypeId: number = undefined;
    negotiationType: NegotiationType;

    @range(0, 2147483647)
    projectModuleId: number = undefined;
    projectModule: ProjectModule;

    @range(0, 2147483647)
    relationshipRequireId: number = undefined;
    relationshipRequire: RelationshipRequire;

    @range(0, 2147483647)
    valueObjectiveId: number = undefined;
    valueObjective: ValueObjective;
    @nested(LongTermObjective)
    longTermObjectives: LongTermObjective[];

    @nested(BackgroundEvent)
    backgroundEvents: BackgroundEvent[];



}

File: RedSheet.DbEntities/Models/ProjectBackground.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
	[Table("ProjectBackgrounds", Schema = "dbo")]
	[Serializable]
	public partial class ProjectBackground  
	{
 
	[Required] 
	[MaxLength(250)]
	public string Focus { get; set; } 
 
	[Required] 
	[MaxLength(100)]
	public string OpponentName { get; set; } 
	///<summary>
    ///{{Status}}
    ///</summary> 
	[DatabaseGenerated(DatabaseGeneratedOption.Identity)] 
	[Key]
	public int ProjectBackgroundId { get; set; } 
 
	[Required] 
	[MaxLength(250)]
	public string Reason { get; set; } 
 
	[RelationshipTableAttribue("NegotiationTypes","dbo")]
	//Relationships
	public int NegotiationTypeId { get; set; } 
 
	[ForeignKey("NegotiationTypeId")]
	public NegotiationType NegotiationType { get; set; } 
 
	[RelationshipTableAttribue("ProjectModules","dbo")]
	//Relationships
	public int ProjectModuleId { get; set; } 
 
	[ForeignKey("ProjectModuleId")]
	public ProjectModule ProjectModule { get; set; } 
 
	[RelationshipTableAttribue("RelationshipRequires","dbo")]
	//Relationships
	public int RelationshipRequireId { get; set; } 
 
	[ForeignKey("RelationshipRequireId")]
	public RelationshipRequire RelationshipRequire { get; set; } 
 
	[RelationshipTableAttribue("ValueObjectives","dbo")]
	//Relationships
	public int ValueObjectiveId { get; set; } 
 
	[ForeignKey("ValueObjectiveId")]
	public ValueObjective ValueObjective { get; set; } 
	[InverseProperty("ProjectBackground")]
	public ICollection<LongTermObjective> LongTermObjectives { get; set; } 

	[InverseProperty("ProjectBackground")]
	public ICollection<BackgroundEvent> BackgroundEvents { get; set; } 


	public ProjectBackground()
	{
	this.LongTermObjectives = new HashSet<LongTermObjective>();

	this.BackgroundEvents = new HashSet<BackgroundEvent>();

    }

    }
}

